pkg/controller/certificaterequests/sync_test.go

```diff
@@ -52,6 +52,7 @@ var (
 )
 
 func generateCSR(t *testing.T, secretKey crypto.Signer, alg x509.SignatureAlgorithm) []byte {
+	t.Helper()
```

Comment:
I've never seen this function before. I read the documentation but still not sure what effect it has on the test output.
Can you explain it to me?

Reply:
It basically ensures that if we fail inside that helper function, the test log output will show the line from which the helper was called and not the line that failed in the helper- this is useful as you probably first want to know which bit in the tests failed, not that an unknown invocation of this helper failed. It was useful for me when debugging that issue where the RSA CSR was re-used for the ECDSA test.

See output where the helper fails _without_ the `t.Helper()`:
```

--- FAIL: TestSync (0.12s)
    /home/irbe/workspace/jetstack/cert-manager/pkg/controller/certificaterequests/sync_test.go:55: 
```
^ Prints the line in helper function that failed, but this doesn't help me to understand which part in tests failed as the helper is invoked multiple times from the same test

Output where the helper fails _with_ `t.Helper()`:
```
--- FAIL: TestSync (0.35s)
    /home/irbe/workspace/jetstack/cert-manager/pkg/controller/certificaterequests/sync_test.go:118: 

```
^ I see that the [invocation of this helper on line 118](https://github.com/irbekrm/cert-manager/blob/bump_go_and_deps/pkg/controller/certificaterequests/sync_test.go#L118) failed, perhaps something wrong with input